projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.native.tsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+/**
+ * WordPress dependencies
+ */
+import { SandBox } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { View, Text } from 'react-native';
+/**
+ * Internal dependencies
+ */
+import style from '../../style.scss';
+/**
+ * Types
+ */
+import type { PlayerProps } from './types';
+
+type NativePlayerProps = Pick< PlayerProps, 'html' | 'isRequestingEmbedPreview' | 'isSelected' >;
+
+/**
+ * VideoPlayer react component
+ *
+ * @param {object} props - Component props.
+ * @param {string} props.html - HTML markup for the player.
+ * @param {boolean} props.isRequestingEmbedPreview - Whether the preview is being requested.
+ * @param {boolean} props.isSelected - Whether the block is selected.
+ * @returns {object}                     - React component.
+ */
+export default function Player( {
+	html,
+	isRequestingEmbedPreview,
+	isSelected,
+}: NativePlayerProps ) {
+	// Set up style for when the player is loading.
+	const loadingStyle: { height?: number } = {};
+	if ( ! html || isRequestingEmbedPreview ) {
+		loadingStyle.height = 250;
+	}
+
+	return (
+		<View style={ { ...style[ 'wp-block-jetpack-videopress__video-player' ], ...loadingStyle } }>
+			{ ! isSelected && (
+				<View style={ style[ 'wp-block-jetpack-videopress__video-player-overlay' ] } />
+			) }
+
+			{ ! isRequestingEmbedPreview && <SandBox html={ html } /> }
+			{ ! html && <Text>{ __( 'Loading…', 'jetpack-videopress-pkg' ) }</Text> }
```

Comment:
I intentionally left out the caption for now. I'll add that in a following PR

Reply:
I couldn't see the `Loading...` when testing. Probably this is caused by the fact that the block renders a black background ([reference](https://github.com/Automattic/jetpack/blob/5cc113fe9e9d90e753b9b6098d1ccbff5527f4de/projects/packages/videopress/src/client/block-editor/blocks/video/style.native.scss#L4)). Not sure if this background was set for testing purposes, but I'd advocate using a transparent color, WDYT?

Additionally, I noticed that for some reason videos have an extra space at the bottom. This might be expected as I can reproduce it in the web version. Not sure if it's an issue in the embedded player, but this enforces the approach of using a transparent color for the background.

Web|App
--|--
<img width="545" alt="Screenshot 2023-03-21 at 12 24 24" src="https://user-images.githubusercontent.com/14905380/226592555-dc4acef5-6bbf-4c05-893b-d8126f5c5f5f.png">|<img width="390" alt="Screenshot 2023-03-21 at 12 25 07" src="https://user-images.githubusercontent.com/14905380/226592567-62be19f8-5d32-4da0-ad60-1b781a0d339b.png">

cc @SiobhyB

Reply:
> I couldn't see the `Loading...` when testing. Probably this is caused by the fact that the block renders a black background ([reference](https://github.com/Automattic/jetpack/blob/5cc113fe9e9d90e753b9b6098d1ccbff5527f4de/projects/packages/videopress/src/client/block-editor/blocks/video/style.native.scss#L4)). Not sure if this background was set for testing purposes, but I'd advocate using a transparent color, WDYT?

Ah correct, I believe that was added back when the player was just a place holder, I'll remove it.

I noticed there is some "corrective" padding in the web component but it looks like it's only used for the initial height calculation i.e. before the video is loaded. Removing the black background will help but I can look into correcting that extra spacing in a follow up PR.

Reply:
fixed in 07d498f468adca0f91199ecf7eba8aac10218a0c